plugins/node/opentelemetry-koa-instrumentation/test/koa-router.test.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { context } from '@opentelemetry/api';
+import { NoopLogger } from '@opentelemetry/core';
+import { NodeTracerProvider } from '@opentelemetry/node';
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks';
+import {
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/tracing';
+import * as assert from 'assert';
+import * as koa from 'koa';
+import * as KoaRouter from '@koa/router';
+import * as http from 'http';
+import { AddressInfo } from 'net';
+import { plugin } from '../src';
+import { AttributeNames, KoaLayerType, KoaComponentName } from '../src/types';
+
+const httpRequest = {
+  get: (options: http.ClientRequestArgs | string) => {
+    return new Promise((resolve, reject) => {
+      return http.get(options, resp => {
+        let data = '';
+        resp.on('data', chunk => {
+          data += chunk;
+        });
+        resp.on('end', () => {
+          resolve(data);
+        });
+        resp.on('error', err => {
+          reject(err);
+        });
+      });
+    });
+  },
+};
+
+describe('Koa Instrumentation - Router Tests', () => {
+  const logger = new NoopLogger();
+  const provider = new NodeTracerProvider();
+  const memoryExporter = new InMemorySpanExporter();
+  const spanProcessor = new SimpleSpanProcessor(memoryExporter);
+  provider.addSpanProcessor(spanProcessor);
+  const tracer = provider.getTracer('default');
+  let contextManager: AsyncHooksContextManager;
+
+  before(() => {
+    plugin.enable(koa, provider, logger);
+  });
+
+  beforeEach(() => {
+    contextManager = new AsyncHooksContextManager();
+    context.setGlobalContextManager(contextManager.enable());
+  });
+
+  afterEach(() => {
+    memoryExporter.reset();
+    context.disable();
+  });
+
+  describe('Instrumenting @koa/router calls', () => {
+    it('should create a child span for middlewares', async () => {
+      const rootSpan = tracer.startSpan('rootSpan');
+      const app = new koa();
+      app.use((ctx, next) => tracer.withSpan(rootSpan, next));
+
+      const router = new KoaRouter();
+      router.get('/post/:id', ctx => {
+        const id = ctx.params.id;
+        ctx.body = 'Post id: ' + id;
```

Comment:
nit: Rather than string concatenation you can use [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

Reply:
Will fix!